File: 16_3Sum Closest.py
class Solution:
    def threeSumClosest(self, s: List[int], target: int) -> int:
        l=len(s)
        op=[]
        for i in range(l-2):
            j=i+1
            while j<(l-1):
                k=j+1
                while k<(l):
                    sum=s[i]+s[j]+s[k]
                    op.append(sum)
                    
                    k=k+1
                j=j+1
        result=[]
        print(op)
        for i in range(len(op)):
            a=target-op[i]
            result.append(a)
        print(result)
        re=[abs(num) for num in result]
        r=0
        for i in range(len(result)):
            if re[i]==min(re):
                r=i
        
        return op[r]
            
            
            
        
       
        

File: 16_3Sum_Closest.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) {
        if (nums.size()<3)
            return 0;
        int result;
        int diff=INT_MAX;
        sort(nums.begin(),nums.end());
        for(int i=0;i<nums.size()-1;i++)
        {
          int left=i+1;
          int right=nums.size()-1;
            
            while(left<right)
            {
                int sum=nums[i]+nums[left]+nums[right];
                if(sum-target==0)
                    return target;
                diff=min(diff,abs(sum-target));
                
                    if(diff==abs(sum-target))
                    {
                        result=sum;
                        
                    }
                
                if(sum<target)
                    left++;
                if(sum>target)
                    right--;
                
                
            }
             
            
            
        }
        return result;
        
    }
};